module/VuFind/tests/integration-tests/src/VuFindTest/Mink/ShibbolethLogoutNotificationTest.php

```diff
@@ -0,0 +1,111 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+namespace VuFindTest\Mink;
+
+use VuFind\Db\Table\ExternalSession;
+
+/**
+ * Shibboleth logout notification test class.
+ *
+ * Class must be final due to use of "new static()" by LiveDatabaseTrait.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ * @retry    4
+ */
+final class ShibbolethLogoutNotificationTest
+    extends \VuFindTest\Integration\MinkTestCase
+{
+    use \VuFindTest\Feature\FixtureTrait;
+    use \VuFindTest\Feature\LiveDatabaseTrait;
+    use \VuFindTest\Feature\LiveDetectionTrait;
+
+    /**
+     * Standard setup method.
+     *
+     * @return void
+     */
+    public function setUp(): void
+    {
+        // Give up if we're not running in CI:
+        if (!$this->continuousIntegrationRunning()) {
+            $this->markTestSkipped('Continuous integration not running.');
+            return;
+        }
+    }
+
+    /**
+     * Test Shibboleth logout notification.
+     *
+     * @return void
+     */
+    public function testLogoutNotification()
+    {
+        $this->changeConfigs(
+            [
+                'permissions' => [
+                    'api.ShibbolethLogoutNotification' => [
+                        'permission' => 'access.api.ShibbolethLogoutNotification',
+                        'require' => 'ANY',
+                        'ipRange' => [
+                            '127.0.0.1',
+                            '::1',
+                        ],
+                    ]
+                ]
+            ]
+        );
+
+        // Do a search and make sure it's in history:
+        $page = $this->performSearch('building:weird_ids.mrc');
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Search/History');
+        $this->findCss($page, 'table#recent-searches');
+
+        // Add a session id mapping to external_session table:
+        $sessionId = $session->getCookie('VUFIND_SESSION');
+        $table = $this->getTable(ExternalSession::class);
+        $table->addSessionMapping($sessionId, 'EXTERNAL_SESSION_ID');
+
+        // Call the notification endpoint:
+        $http = new \VuFindHttp\HttpService();
+        $result = $http->post(
```

Comment:
Would it make sense to do this post twice -- once before updating the database and again after? That way we can confirm that it returns a bad status when the data is missing, and a good status when the data is present.

Reply:
It won't reveal whether the request was successful or not, but returns OK always.

Reply:
Ahh, okay, I see -- so it's validating the emptiness of the search history that confirms that it actually did something. I understand now!